oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -132,6 +132,9 @@ private PropertySource oktaStaticDiscoveryPropertySource(Environment environment
         Map<String, Object> properties = new HashMap<>();
         properties.put("spring.security.oauth2.resourceserver.jwt.issuer-uri", "${okta.oauth2.issuer}");
         properties.put("spring.security.oauth2.resourceserver.jwt.jwk-set-uri", "${okta.oauth2.issuer}/v1/keys");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.client-id", "${okta.oauth2.clientId}");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.client-secret", "${okta.oauth2.clientSecret}");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.introspection-uri", "${okta.oauth2.introspectionUri");
```

Comment:
These need to be conditional on `okta.oauth2.clientSecret` being set

Reply:
Sorry, did you mean conditional on `okta.oauth2.opaque` being set instead?

Reply:
No, assuming there is no downside to configuring both. I'm just not sure what happens when you set the introspection-uri and ID but no secret is used (I assume that would fail).
The `okta.oauth2.opauqe` flag would just trigger the default SecurityConfigurer to use opaque or jwt (currently the default is to always use JWT)

NOTE, there could be other conditionals here that I'm not thinking about, so we need to make sure we have good testing around this